python/rpdk/java/templates/init/guided_aws/StubCreateHandler.java

```diff
@@ -29,157 +26,92 @@
 
         this.logger = logger;
 
-        final ResourceModel model = request.getDesiredResourceState();
-
         // TODO: Adjust Progress Chain according to your implementation
         // https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/blob/master/src/main/java/software/amazon/cloudformation/proxy/CallChain.java
 
-        return ProgressEvent.progress(model, callbackContext)
+        return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
 
             // STEP 1 [check if resource already exists]
             // if target API does not support 'ResourceAlreadyExistsException' then following check is required
             // for more information -> https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract.html
-            //.then(progress -> checkForPreCreateResourceExistence(proxy, request, progress))
+            .then(progress ->
+                // STEP 1.0 [initialize a proxy context]
+                // If your service API is not idempotent, meaning it does not distinguish duplicate create requests against some identifier (e.g; resource Name)
+                // and instead returns a 200 even though a resource already exists, you must first check if the resource exists here
+                // NOTE: If your service API throws 'ResourceAlreadyExistsException' for create requests this method is not necessary
+                proxy.initiate("{{ call_graph }}::{{ operation }}::PreExistanceCheck", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+
+                    // STEP 1.1 [TODO: construct a body of a request]
+                    .translateToServiceRequest(Translator::translateToReadRequest)
+
+                    // STEP 1.2 [TODO: make an api call]
+                    .makeServiceCall((awsRequest, client) -> {
+                        AwsResponse awsResponse = null;
+
+                        // TODO: add custom read resource logic
+
+                        logger.log(String.format("%s has successfully been read.", ResourceModel.TYPE_NAME));
+                        return awsResponse;
+                    })
+
+                    // STEP 1.3 [TODO: handle exception]
+                    .handleError((awsRequest, exception, client, model, context) -> {
+                        // TODO: uncomment when ready to implement
+                        // if (exception instanceof CfnNotFoundException)
+                        //     return ProgressEvent.progress(model, context);
+                        // throw exception;
```

Comment:
I tested throwing exceptions here, and they are not bubbled all the way up. The handler ends up returning a status code of FAILED but no exception is thrown.

Reply:
The intent is the Handler should return FAILED with a sane/consistent exception message that ends up in Stack Trace. I think there's a separate issue that needs to be fixed for that as it's not working reliably.

Reply:
So which of these asserts should I use?

```java
assertThatThrownBy(() -> handler.handleRequest(proxy, request, new CallbackContext(), proxyClient, logger))
 .isInstanceOf(CfnAlreadyExistsException.class);
```
or

```java
response = handler.handleRequest(proxy, request, new CallbackContext(), proxyClient, logger);
assertThat(response).isNotNull();
assertThat(response.getStatus()).isEqualTo(OperationStatus.FAILED);
assertThat(response.getErrorCode()).isEqualTo(HandlerErrorCode.AlreadyExists);
```

[I've seen plenty of examples of the former](https://github.com/search?q=assertThrows+filename%3Acreatehandlertest.java+org%3Aaws-cloudformation&type=Code) but according to your suggestion we should use the latter?